clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -730,19 +768,22 @@ raw2trace_t::do_conversion()
 }
 
 raw2trace_t::block_summary_t *
-raw2trace_t::lookup_block_summary(void *tls, app_pc block_start)
+raw2trace_t::lookup_block_summary(void *tls, uint64 modidx, uint64 modoffs,
+                                  app_pc block_start)
 {
     auto tdata = reinterpret_cast<raw2trace_thread_data_t *>(tls);
-    if (block_start == tdata->last_decode_block_start) {
+    if (block_start == tdata->last_decode_block_start &&
+        modidx == tdata->last_decode_modidx && modoffs == tdata->last_decode_modoffs) {
```

Comment:
Because there is no sentinel value with modidx:modoffs that is not a valid value so we can't tell whether this is initialized or not.  There is a comment explaining this by this field in the header.

Reply:
Oh okay. I missed that comment.